catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -73,10 +69,33 @@ public MetacardTypeImpl(String name, Set<AttributeDescriptor> descriptors) {
          * serialized object is maintained. For instance, if a null check is added in the
          * constructor, the same check should be added in the readObject() method.
          */
-        this.name = name;
-        if (descriptors != null) {
-            this.descriptors.addAll(descriptors);
+        this(null, descriptors, name);
+    }
+
+    /**
+     * Creates a {@code MetacardTypeImpl} with the provided {@link MetacardType}, set of
+     * additional {@linkplain AttributeDescriptor AttributeDescriptors}, and name.
+     * <p>
+     * {@code additionalDescriptors} and the descriptors in {@code metacardType} will be combined to
+     * form the set of descriptors for this {@code MetacardTypeImpl}.
+     * <p>
+     * Essentially, this constructor provides an easy way to create a new {@code MetacardTypeImpl}
+     * that extends an existing {@link MetacardType}.
+     *
+     * @param metacardType          the base {@link MetacardType}
+     * @param additionalDescriptors the descriptors to add to this {@code MetacardTypeImpl} in
+     *                              addition to the descriptors in {@code metacardType}
+     * @param name                  the name of this {@code MetacardTypeImpl}
+     */
+    public MetacardTypeImpl(MetacardType metacardType,
+            Set<AttributeDescriptor> additionalDescriptors, String name) {
```

Comment:
Actually it probably doesn't make sense to allow a null or empty set because if you did, you'd just be creating a duplicate metacard type.